mat/io.go

```diff
@@ -370,3 +427,53 @@ func readFull(r io.Reader, buf []byte) (int, error) {
 	}
 	return n, err
 }
+
+// storage is the internal representation of the storage format of a
+// serialised matrix.
+type storage struct {
+	Version uint32 // Keep this first.
+	Form    byte   // [GST]
```

Comment:
I still feel that the meaning of the `'G'`, `'S'`, `'T'`, ... letters could be explained somewhere.
presumably:
- `G` -> General
- `S` -> Symmetric
- `T` -> Triangular
- `B` -> Band ?
- `P` -> Packed ?
- `F` -> Fortran?
- `A` -> Any?
- `U` -> Up
- `L` -> Low

Reply:
These are in https://github.com/gonum/gonum/pull/374#discussion_r187238701. I can put that text in the source.

Full,Band,Packed
All,Upper,Lower.

Reply:
Added table of scheme.